src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/VerifyClientDeviceIdentityTest.java

```diff
@@ -234,16 +232,17 @@ void GIVEN_broker_WHEN_verify_client_identity_with_internal_server_exception_THE
                     .withCredential(new ClientDeviceCredential()
                             .withClientDeviceCertificate("abc"));
 
-            Exception err = Assertions.assertThrows(Exception.class, () -> {
-                verifyClientIdentity(ipcClient, request, null);
-            });
-            assertEquals(err.getCause().getClass(), ServiceError.class);
+            Pair<CompletableFuture<Void>, Consumer<VerifyClientDeviceIdentityResponse>> cb =
+                    asyncAssertOnConsumer((m) -> assertFalse(m.isIsValidClientDevice()));
+
+            verifyClientIdentity(ipcClient, request, cb.getRight());
+            cb.getLeft().get(10, TimeUnit.SECONDS);
```

Comment:
I think this is more complicated than it needs to be. There's no need for anything to be async. `verifyClientIdentity` already blocks.

```
Consumer<VerifyClientDeviceIdentityResponse> cb = r -> assertFalse(r.isIsValidClientDevice());
verifyClientIdentity(ipcClient, request, cb)
```